### DISCUSSION TITLE
"Internal Server Error"

### DISCUSSION THREAD

**Discussion by TechnicalSuwako** [searx/searx#2975] [Installation issue: Official wiki] [answered]
**Version of Searx, commit number if you are using on master branch and stipulate if you forked Searx**
commit e88bea53ebf220648685cdb8eff017357728494e
1.0.0

**How did you install Searx?**
usual

**What happened?**
When I access the server, I get an "Internal Server Error".

**How To Reproduce**
https://search.076.ne.jp/searx

**Expected behavior**
What can be used

**Screenshots & Logs**

```
root@search  /opt/searx   master ±  sudo -H ./utils/searx.sh inspect service

service status & log
====================

sourced .config.sh :

  PUBLIC_URL          : http://search.076.ne.jp/searx
  SEARX_URL_PATH      : /searx
  SEARX_INSTANCE_NAME : searx@search.076.ne.jp 
  SEARX_INTERNAL_HTTP  : 127.0.0.1:8888

INFO:  Service account searx exists.
INFO:  ~searx: python environment is available.
INFO:  ~searx: Searx software is installed.
INFO:  uWSGI app searx.ini is enabled.
INFO:  public URL   --> http://search.076.ne.jp/searx
INFO:  internal URL --> http://127.0.0.1:8888
INFO:  got 500 from http://127.0.0.1:8888
INFO:  got 429 from http://search.076.ne.jp/searx

Enable searx debug mode? [NO/yes] y
WARN:  Do not enable debug in production enviroments!!
INFO:  try to enable debug mode ...
cd /usr/local/searx/searx-src
sed -i -e "s/debug : False/debug : True/g" "/etc/searx/settings.yml"
INFO:  restart uWSGI service

Unit searx.service could not be found.
// use CTRL-C to stop monitoring the log
Sat Sep 18 09:26:58 2021 - unable to load app 0 (mountpoint='') (callable not found or import error)
Sat Sep 18 09:26:58 2021 - *** no app loaded. going in full dynamic mode ***
Traceback (most recent call last):
  File "/usr/local/searx/searx-src/searx/__init__.py", line 19, in <module>
    import searx.settings_loader
  File "/usr/local/searx/searx-src/searx/settings_loader.py", line 8, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
Sat Sep 18 09:26:58 2021 - unable to load app 0 (mountpoint='') (callable not found or import error)
Sat Sep 18 09:26:58 2021 - *** no app loaded. going in full dynamic mode ***
^CINFO:  try to disable debug mode ...
cd /usr/local/searx/searx-src
sed -i -e "s/debug : True/debug : False/g" "/etc/searx/settings.yml"
INFO:  restart uWSGI service
```

```
root@search  /opt/searx   master ±  cat /var/log/uwsgi/app/searx.log

...
Sat Sep 18 09:27:04 2021 - spawned uWSGI http 1 (pid: 10876)
Traceback (most recent call last):
  File "/usr/local/searx/searx-src/searx/__init__.py", line 19, in <module>
    import searx.settings_loader
  File "/usr/local/searx/searx-src/searx/settings_loader.py", line 8, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
Sat Sep 18 09:27:04 2021 - unable to load app 0 (mountpoint='') (callable not found or import error)
Sat Sep 18 09:27:04 2021 - *** no app loaded. going in full dynamic mode ***
Traceback (most recent call last):
  File "/usr/local/searx/searx-src/searx/__init__.py", line 19, in <module>
    import searx.settings_loader
  File "/usr/local/searx/searx-src/searx/settings_loader.py", line 8, in <module>
    import yaml
ModuleNotFoundError: No module named 'yaml'
Sat Sep 18 09:27:04 2021 - unable to load app 0 (mountpoint='') (callable not found or import error)
Sat Sep 18 09:27:04 2021 - *** no app loaded. going in full dynamic mode ***
```

```
root@search  /opt/searx   master ±  pip show pyyaml                  
Name: PyYAML
Version: 5.4.1
Summary: YAML parser and emitter for Python
Home-page: https://pyyaml.org/
Author: Kirill Simonov
Author-email: xi@resolvent.net
License: MIT
Location: /usr/local/lib/python3.9/dist-packages
Requires: 
Required-by:
```

**Comment by TechnicalSuwako** [comment, marked as answer]
I fixed.

**Reply by Grooty12** [reply]
How?

**Reply by sdfg2** [reply]
> I fixed.

Yeah, how?

**Comment by 0xd13ad10c** [comment]
[https://searx.github.io/searx/admin/installation-searx.html](url)

Following the step by step guide above and restarting `uwsgi@searx.service` as well as my apache server at `httpd.service`  solved the issue for me.

I'm thinking it has something to do with pyyaml not being properly installed in the virtual environment for user searx, OR changes needing to be made in `/usr/local/searx/.profile` during the dependencies installation step in the process. 

Whatever it is, it's working now.

**Reply by Grooty12** [reply]
I followed the link and now it works. Weird.
Oh well now it works

**Reply by sdfg2** [reply]
> [https://searx.github.io/searx/admin/installation-searx.html](url)
> 
> Following the step by step guide above and restarting `uwsgi@searx.service` as well as my apache server at `httpd.service` solved the issue for me.
> 
> I'm thinking it has something to do with pyyaml not being properly installed in the virtual environment for user searx, OR changes needing to be made in `/usr/local/searx/.profile` during the dependencies installation step in the process.
> 
> Whatever it is, it's working now.

404 on that page?

**Reply by Grooty12** [reply]
404 on this page: http://127.0.0.1:8888/
![ksnip_20220107-172545](https://user-images.githubusercontent.com/85930101/148574495-188508ce-bbd7-4a8a-9502-e6d0a4f1d09e.png)

**Reply by sdfg2** [reply]
> [https://searx.github.io/searx/admin/installation-searx.html](url)
> 
> Following the step by step guide above and restarting `uwsgi@searx.service` as well as my apache server at `httpd.service` solved the issue for me.
> 
> I'm thinking it has something to do with pyyaml not being properly installed in the virtual environment for user searx, OR changes needing to be made in `/usr/local/searx/.profile` during the dependencies installation step in the process.
> 
> Whatever it is, it's working now.

Ah, the link is different from the text of the link.

**Reply by 0xd13ad10c** [reply]
> > [https://searx.github.io/searx/admin/installation-searx.html](url)
> > Following the step by step guide above and restarting `uwsgi@searx.service` as well as my apache server at `httpd.service` solved the issue for me.
> > I'm thinking it has something to do with pyyaml not being properly installed in the virtual environment for user searx, OR changes needing to be made in `/usr/local/searx/.profile` during the dependencies installation step in the process.
> > Whatever it is, it's working now.
> 
> 404 on that page?

https://searx.github.io/searx/admin/installation-searx.html#installation-basic

This link should work.
